src/nvim/buffer_defs.h

```diff
@@ -968,18 +980,20 @@ typedef struct {
   bool noautocmd;
   bool fixed;
   bool hide;
-} FloatConfig;
-
-#define FLOAT_CONFIG_INIT ((FloatConfig){ .height = 0, .width = 0, \
-                                          .bufpos = { -1, 0 }, \
-                                          .row = 0, .col = 0, .anchor = 0, \
-                                          .relative = 0, .external = false, \
-                                          .focusable = true, \
-                                          .zindex = kZIndexFloatDefault, \
-                                          .style = kWinStyleUnused, \
-                                          .noautocmd = false, \
-                                          .hide = false, \
-                                          .fixed = false })
+} WinConfig;
```

Comment:
The previous naming is cleaner and more intuitive. For a normal window it only requires a few fields. I suggest separate (not sure)

Reply:
@justinmk thoughts since you originally suggested the name change? I'm neutral on this, though I don't think a separate struct would be needed if that's what you mean.

Also, the `vertical` property may be able to be removed from here as it's not actually saved, just "translated" into a value of `split`.

Reply:
this will initialize more fields when split a normal window through `nvim_open_win` :thinking:

Reply:
The fields are initialized either way as far as I know, `win_alloc` initializes `win.w_config` to `WIN_CONFIG_INIT`

Reply:
oh yes.

Reply:
I removed `vertical` from `WinConfig` since it doesn't need to be saved, so the only additional property being saved is the `WinSplit` enum.